recipes/7zip/19.00/conanfile.py

```diff
@@ -41,7 +41,12 @@ def _uncompress_7z(self, filename):
     def source(self):
         from six.moves.urllib.parse import urlparse
         url = self.conan_data["sources"][self.version]["url"]
-        filename = os.path.basename(urlparse(url).path)
+
+        first_url = url[0] if isinstance(url, list) else url
```

Comment:
🤔 Is it better to fail when it's a list or quiet ignore the list? Maybe this can iterate over the list so it works like mirrors as inteded?

Reply:
The problem is, that the conan-own **get** function isn't used because 7z unpack is not part of it.
That's why it has been originally separated.

Download is capable of lists, but for the uncompressing a filename is required. Maybe faking the name would be another choice.

Reply:
And for better... internally we need to hack our own url to download for corporate proxy reasons.